src/test/scala/ChiselSpec.scala

```diff
@@ -22,40 +23,27 @@ import java.io.{ByteArrayOutputStream, PrintStream}
 import scala.reflect.ClassTag
 
 /** Common utility functions for Chisel unit tests. */
-trait ChiselRunners extends Assertions {
-  def runTester(
-    t:                    => BasicTester,
-    additionalVResources: Seq[String] = Seq()
-//                 annotations:          AnnotationSeq = Seq()
-  ): Boolean = {
-    val defaultBackend = chisel3.testers.TesterDriver.defaultBackend
-//    val hasBackend = TestUtils.containsBackend(annotations)
-//    val annos: Seq[Annotation] = if (hasBackend) annotations else defaultBackend +: annotations
-    TesterDriver.execute(() => t, additionalVResources)
+trait ChiselRunners extends TestSuite with Assertions with ChiselSim {
+  def runTester(t: => Module, maxCycles: Int = 1000): Unit = {
+    simulate(t)(RunUntilFinished(maxCycles))
   }
-  def assertTesterPasses(
-    t:                    => BasicTester,
-    additionalVResources: Seq[String] = Seq()
-//                          annotations:          AnnotationSeq = Seq()
-  ): Unit = {
-    assert(runTester(t, additionalVResources /*, annotations*/ ))
+  def assertTesterPasses(t: => Module, maxCycles: Int = 1000): Unit = {
+    runTester(t)
   }
-  def assertTesterFails(
-    t:                    => BasicTester,
-    additionalVResources: Seq[String] = Seq(),
-    annotations:          Seq[chisel3.aop.Aspect[_]] = Seq()
-  ): Unit = {
-    assert(!runTester(t, additionalVResources /*, annotations*/ ))
+
+  /** Users of this method should ensure an exception is thrown, such as with `thrownBy`. */
+  def assertTesterFails(t: => Module, maxCycles: Int = 1000): Unit = {
+    runTester(t)
   }
```

Comment:
This now, as written, is the same as `assertTesterPasses` and isn't actually checking for failure. I'd suggest replace usages of `assertTesterFails` with `intercept[Exception] { ??? }.getMessage should be ("hello")`.

Reply:
Addressed in https://github.com/ucb-bar/fixedpoint/pull/11/commits/28d70785628c5784c10a280c926e7d02549c4652.